### PR TITLE
More clang tidy checks

### PR DESCRIPTION
**Pull request by mkow**
Changes:
* Enabled much more clang-tidy checks
* Fixed a bug in setting C++ standard in CMakeLists
* Fixed a bug saveDataToFile (it didn't work on offsets > 2GB)
* Fixed a bug in kaitai streams (mistaken arguments order when calling a function)

### CODE REVIEW

File: src/visualization/minimap.cc [hunk 1]
```diff
@@ -355,7 +357,7 @@ void VisualizationMinimap::initTextures() {
   texture_->allocateStorage();
 
   point_size_ = std::max(1.0, static_cast<double>(sample_size_) / texture_size);
-  const uint8_t* rowdata = reinterpret_cast<const uint8_t*>(sampler_->data());
+  auto* rowdata = reinterpret_cast<const uint8_t*>(sampler_->data());
```
Comment (catsuryuu) [inline src/visualization/minimap.cc]: const?

File: src/visualization/trigram.cc [hunk 2]
```diff
@@ -436,8 +429,8 @@ void TrigramWidget::initShaders() {
 }
 
 void TrigramWidget::initTextures() {
-  int size = static_cast<int>(getDataSize());
-  const uint8_t* data = reinterpret_cast<const uint8_t*>(getData());
+  auto size = static_cast<int>(getDataSize());
+  auto* data = reinterpret_cast<const uint8_t*>(getData());
```
Comment (catsuryuu) [inline src/visualization/trigram.cc]: const?

File: src/visualization/minimap.cc (hunk 1 above)
Comment (catsuryuu) [reply src/visualization/minimap.cc]: `rowdata` -> `row_data`

File: CMakeLists.txt [hunk 3]
```diff
@@ -38,12 +38,14 @@ if(MINGW)
...
 endif()
-set(CMAKE_CXX_STANDARD_REQUIRED 14) # Use C++14
+set(CMAKE_CXX_STANDARD 14) # Use C++14
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 set(CMAKE_INSTALL_RPATH ".")
 if(CMAKE_COMPILER_IS_GNUCXX)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++14") # All warnings
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")  # All warnings
```
[the remark this answers is not in the document]
Comment (mkow) [reply CMakeLists.txt]: There's no such thing as '+=' in CMake.

File: src/network/msgpackobject.cc [hunk 4]
```diff
@@ -494,7 +494,9 @@ std::shared_ptr<MsgpackObject> toMsgpackObject(
 
 std::shared_ptr<MsgpackObject> toMsgpackObject(
     const std::shared_ptr<proto::VelesException>& val) {
```
[the remark this answers is not in the document]
Comment (mkow) [reply src/network/msgpackobject.cc]: Fixed.

File: src/ui/fileblobmodel.cc [hunk 5]
```diff
@@ -187,13 +186,12 @@ QModelIndex FileBlobModel::index(int row, int column,
   return createIndex(row, column, itemFromIndex(parent));
 }
 
-QModelIndex FileBlobModel::parent(const QModelIndex& index) const {
-  if (!index.isValid()) {
-    return QModelIndex();
+QModelIndex FileBlobModel::parent(const QModelIndex& child) const {
```
[the remark this answers is not in the document]
Comment (mkow) [reply src/ui/fileblobmodel.cc]: Fixed

File: src/ui/veles_mainwindow.cc [hunk 6]
```diff
@@ -47,9 +47,7 @@ using util::settings::shortcuts::ShortcutsModel;
 /*****************************************************************************/
 
 VelesMainWindow::VelesMainWindow()
-    : MainWindowWithDetachableDockWidgets(),
```
[the remark this answers is not in the document]
Comment (mkow) [reply src/ui/veles_mainwindow.cc]: Yes, base classes constructors are always called first even if you not specify them here. This syntax can be used to call a non-default constructor (e.g. one with arguments).

File: src/visualization/digram.cc [hunk 7]
```diff
@@ -82,15 +85,15 @@ void DigramWidget::initTextures() {
...
+  const auto* rowdata = reinterpret_cast<const uint8_t*>(getData());
   for (size_t i = 0; i < getDataSize() - 1; i++) {
     size_t index = rowdata[i] * 512 + rowdata[i + 1] * 2;
     bigtab[index]++;
     bigtab[index + 1] += i;
   }
   for (int i = 0; i < 256; i++) {
     for (int j = 0; j < 256; j++) {
-      size_t index = i * 512 + j * 2;
+      int index = i * 512 + j * 2;
```
[the remark this answers is not in the document]
Comment (mkow) [reply src/visualization/digram.cc]: Yes, this index always fits in an `int` and is quite small, so we can use the most popular type in C++ for small numbers (int) ;)
In the previous loop we iterate up to getDataSize(), which can return something large.

File: src/visualization/minimap.cc (hunk 1 above)
Comment (mkow) [reply src/visualization/minimap.cc]: this is already const, no need for adding it (same for '*', but IMO it increases readability).
changed to `row_data`

File: src/visualization/trigram.cc (hunk 2 above)
Comment (mkow) [reply src/visualization/trigram.cc]: ditto

**Pull request merged**